src/smt/command.cpp

```diff
@@ -2139,6 +2139,91 @@ std::string GetSynthSolutionCommand::getCommandName() const
   return "get-instantiations";
 }
 
+GetInterpolCommand::GetInterpolCommand() {}
+GetInterpolCommand::GetInterpolCommand(const std::string& name, Expr conj)
+    : d_name(name), d_conj(conj), d_resultStatus(false)
+{
+}
+GetInterpolCommand::GetInterpolCommand(const std::string& name,
+                                       Expr conj,
+                                       const Type& gtype)
+    : d_name(name),
+      d_conj(conj),
+      d_sygus_grammar_type(gtype),
+      d_resultStatus(false)
+{
+}
+
+Expr GetInterpolCommand::getConjecture() const { return d_conj; }
+Type GetInterpolCommand::getGrammarType() const { return d_sygus_grammar_type; }
+Expr GetInterpolCommand::getResult() const { return d_result; }
+
+void GetInterpolCommand::invoke(SmtEngine* smtEngine)
```

Comment:
It would be good if this already used the new API instead of the deprecated old API (which will be removed shortly). For an example, see the `DefineFunctionRecCommand`: https://github.com/CVC4/CVC4/blob/master/src/smt/command.h#L509

Reply:
Good point. Could you give me more clue on how to do this? I tried to add `api::` in front of `Expr` or `SmtEngine` etc, but got errors of "has not been declared".

Reply:
`Expr` is part of the old API, the new API (in `cvc4cpp.h`) wraps `Expr` in `Term`. So what you need is `api::Solver` instead of `SmtEngine`, and then you call `Solver::getInterpol` instead of `SmtEngine::getInterpol`. We can discuss this more offline if you have more questions.